# de Mattia & Ruhlmann-Kleider (2019) 的完整 GIC 与我们的 2PCF 模型

日期：2026-07-12  
目的：用尽量直白的语言回答三个问题：论文到底算了什么、我们现在算了什么、两者之间少了什么。

参考文献：A. de Mattia & V. Ruhlmann-Kleider, Integral constraints in spectroscopic surveys, JCAP 08 (2019) 036, [arXiv:1904.08851](https://arxiv.org/abs/1904.08851)。

> **一句话结论：我们当前的 parameter-dependent W2 修正正确计算了窗口平均密度方差 C=σW²，但模型 ξ−C 是论文完整结果 ξ−2B(s)+C 的单项近似；Task4.3 还因为从数据重采样 random redshift 而额外包含了尚未建模的 radial integral constraint（RIC）。**

## 0. 先不要被公式吓到：问题是怎么产生的？

真实宇宙有一个真正的平均密度，但有限巡天并不知道它。我们只能用巡天内部的数据估计平均密度。因此，巡天内部看见的平均过密度会被强制设为零。

$$
\bar\delta_W \equiv \int d^3y\,W_N(\mathbf y)\delta(\mathbf y),
\qquad
\delta_{\rm obs}(\mathbf x)=\delta(\mathbf x)-\bar\delta_W,
$$

其中 $W_N$ 是归一化后的 survey selection function，满足 $\int W_N=1$。

逻辑流程可以写成：

$$
\text{真实场 }\delta
\;\longrightarrow\;
\text{用本巡天估计平均值 }\bar\delta_W
\;\longrightarrow\;
\delta_{\rm obs}=\delta-\bar\delta_W
\;\longrightarrow\;
\text{二点函数出现 4 项。}
$$

## 1. 最关键的一步：把两个括号乘开

观测二点函数来自

$$
\left\langle
[\delta_1-\bar\delta_W]
[\delta_2-\bar\delta_W]
\right\rangle.
$$

乘开以后是

$$
\underbrace{\langle\delta_1\delta_2\rangle}_{\xi(s)}
-\underbrace{\langle\delta_1\bar\delta_W\rangle}_{B_1(s)}
-\underbrace{\langle\bar\delta_W\delta_2\rangle}_{B_2(s)}
+\underbrace{\langle\bar\delta_W^2\rangle}_{C}.
$$

四项的直观含义是：

1. $\xi(s)$：真实密度与真实密度的相关；
2. $B_1(s)$：第一个点与巡天平均密度的相关；
3. $B_2(s)$：巡天平均密度与第二个点的相关；
4. $C$：巡天平均密度与自身的相关，也就是 $\sigma_W^2$。

对偶数多极矩和对称 pair average，通常有 $B_1(s)=B_2(s)\equiv B(s)$，所以完整结果是

$$
\boxed{\xi_{\rm full}(s)=\xi(s)-2B(s)+C.}
$$

## 2. 论文中的公式和上面的四项一一对应

de Mattia & Ruhlmann-Kleider (2019) 的式 (2.20) 写成

$$
\xi_\ell^{\rm cic}(s)
=
\xi_\ell^{\rm c}(s)
-IC_\ell^{\delta,{\rm ic}}(s)
-IC_\ell^{{\rm ic},\delta}(s)
+IC_\ell^{{\rm ic},{\rm ic}}(s).
$$

| 论文记号 | 直白含义 | 上一节记号 |
|---|---|---|
| $\xi^{\rm c}$ | 普通相关函数乘 survey window | $\xi$ 对应的 windowed 项 |
| $-IC^{\delta,{\rm ic}}$ | 密度与被减掉的平均模式交叉相关 | $-B_1(s)$ |
| $-IC^{{\rm ic},\delta}$ | 反方向的交叉相关 | $-B_2(s)$ |
| $+IC^{{\rm ic},{\rm ic}}$ | 被减掉的平均模式的自相关 | $+C$ |

论文这里首先计算的是尚未除以 RR 的 window-multiplied correlation。与 Landy–Szalay estimator 比较时，还要除以相同的 $RR(s)$ window。对于真实空间、各向同性 monopole 和很窄的 separation bin，普通的两点 window 大部分会被这个除法消掉；但两个 density–IC 交叉项不会因此自动消失。

## 3. 我们当前算的是什么？

我们当前的模型是

$$
\boxed{\xi_{\rm ours}(s)=\xi(s)-\sigma_W^2(\theta)=\xi(s)-C.}
$$

把它与完整结果比较：

$$
\begin{aligned}
\text{完整：}&\quad \xi-2B(s)+C,\\
\text{当前：}&\quad \xi-C.
\end{aligned}
$$

两者只有在

$$
B(s)\simeq C
$$

时才近似相等，因为此时 $-2B+C\simeq-C$。

所以，我们不是完全没有考虑交叉项的物理作用，而是用一个额外假设，把完整的 $-2B(s)+C$ 整体近似为 $-C$。问题在于：$B(s)$ 一般依赖 separation 和 survey geometry，不一定总等于常数 $C$。

### 3.1 为什么这个近似不是凭空来的？

论文的式 (2.23) 回顾了传统单项近似：

$$
P^{\rm cic}(\mathbf k)
=
P^{\rm c}(\mathbf k)
-P^{\rm c}(\mathbf 0)|\widetilde W(\mathbf k)|^2.
$$

作者明确说明：这个旧公式只使用一个有效修正项，而完整推导有三个修正项。他们也发现，在论文展示的 CMASS-like global-IC 几何中，这个单项近似与完整结果非常接近。

因此正确判断不是“我们的公式毫无依据”，而是：

> **我们的公式是有文献依据的 leading/single-term approximation，但还没有证明它在我们的 cube、positive-octant lightcone、local-PNG 和 50–350 Mpc/h 范围内足够准确。**

## 4. W2(k) 到底算对了什么？

我们的 random-pair kernel 是

$$
W_2(k)
\simeq
\frac{\sum_a RR_a\,\overline{j_0}(k;s_a)}{\sum_a RR_a},
$$

然后计算

$$
C(\theta)=\sigma_W^2(\theta)
=
\int\frac{k^2dk}{2\pi^2}P(k;\theta)W_2(k),
$$

或在 simulation mother-box 中使用相同公式的 FullDiscrete mode sum。

这一部分有三个优点：

1. $C$ 不是随手拟合的常数，而是由理论 $P(k;\theta)$ 与实际 geometry 决定；
2. 它随 $f_{\rm NL}$、$b_1$ 等参数更新；
3. Task4.2 使用解析 cubic window，Task4.3 使用 FKP-weighted random pairs。

但是，一个只依赖 $k$ 的一维 $W_2(k)$ 只能直接给出 $C$。完整的 $B(s)$ 还需要包含 pair separation $s$、内部积分距离 $\Delta$ 和 survey geometry 的更高维 window kernel；这就是当前缺失的信息。

## 5. 直接查看当前代码证据

下面的单元读取源码的关键行。它不修改任何文件。

In [1]:
# 代码大纲：
# 1. 固定项目根目录，确保 notebook 从任意工作目录启动时都能找到源码。
# 2. 定义按行号读取源码的小工具，只打印与 GIC 修正直接有关的局部代码。
# 3. 分别展示 Task4.2 与 Task4.3 中减去 sigma_W^2 的实现，并做字符串断言。

from pathlib import Path

ROOT = Path('/pscratch/sd/l/lzy/PNG-2PCF-model-to-observe')

def show_lines(relative_path: str, first: int, last: int) -> None:
    """按一开始计数的行号打印源码片段，便于把结论追溯到当前实现。"""
    path = ROOT / relative_path
    lines = path.read_text(encoding='utf-8').splitlines()
    print(f'\n[{relative_path}:{first}]')
    for number in range(first, last + 1):
        print(f'{number:4d}: {lines[number - 1]}')

task42_path = 'codes/task4/task4p2_pk_rawbox_subbox/task4p2_fit_2pcf_shellavg_fixed_sn0.py'
task43_path = 'codes/task43/task43_fit_minimal_closure.py'

show_lines(task42_path, 482, 490)
show_lines(task43_path, 443, 455)

task42_text = (ROOT / task42_path).read_text(encoding='utf-8')
task43_text = (ROOT / task43_path).read_text(encoding='utf-8')
assert 'xi = xi - sigma_w2' in task42_text
assert 'return base_xi - formal_sigma_w2(theta)' in task43_text
print('\n检查通过：两条当前主线都执行 parameter-dependent 的单常数减法。')


[codes/task4/task4p2_pk_rawbox_subbox/task4p2_fit_2pcf_shellavg_fixed_sn0.py:482]
 482: def model_xi(case: FitCase, basis: ShellBasis, theta: np.ndarray) -> tuple[np.ndarray, float]:
 483:     """计算一个 case 的 shell-averaged raw/no-GIC/formal-GIC 模型。"""
 484:     c0, c1, c2 = coefficients(theta)
 485:     xi = c0 * basis.xi_pkdd + c1 * basis.xi_gamma_pkdd + c2 * basis.xi_gamma2_pkdd
 486:     sigma_w2 = 0.0
 487:     if case.model_label == "formal_gic":
 488:         sigma_w2 = c0 * basis.sigma_pkdd + c1 * basis.sigma_gamma_pkdd + c2 * basis.sigma_gamma2_pkdd
 489:         xi = xi - sigma_w2
 490:     return np.asarray(xi, dtype="f8"), float(sigma_w2)

[codes/task43/task43_fit_minimal_closure.py:443]
 443:     else:
 444:         raise ValueError(f"unknown xi_kernel: {xi_kernel_mode}")
 445: 
 446:     def xi_from_pk_dense(pk_dense: np.ndarray) -> np.ndarray:
 447:         weight = theory["g_nz"] * task41.interp_logk(theory["k_eff"], theory["k_dense"], pk_dense)
 448:         return (we

## 6. 现有 posterior 到底告诉了我们什么？

下面直接读取正式 summary，而不是手工抄数字。这里的“formal GIC”应理解为当前代码中的 parameter-dependent single-term approximation。

In [2]:
# 代码大纲：
# 1. 读取 Task4.2 当前 shell-averaged summary，提取 no-GIC 与单项 GIC 的 fNL 中位数。
# 2. 读取 Task4.3 保存的 matched no/formal summary，提取相同量和 sigma_W^2。
# 3. 计算 GIC 相对 no-GIC 的中心值位移，并用 Markdown 表格显示。

import json
from IPython.display import Markdown, display

task42_summary_path = ROOT / 'outputs/task4_outputs/task4p2_2pcf_shellavg_fixed_sn0_z1/task4p2_2pcf_shellavg_fixed_sn0_z1_summary.json'
task43_summary_path = ROOT / 'old_doc_codes/task4_task44_cleanup_20260707T061844Z/moved/outputs/task43_outputs/fits/full25_mean_s50_350_ds10_shellavg_besselfftlog_rrdeconv_fitcov_L2000_p1p0_mcmc/task43_minimal_closure_mcmc_summary.json'

task42 = json.loads(task42_summary_path.read_text(encoding='utf-8'))['summaries']
task43_models = {
    row['model']: row
    for row in json.loads(task43_summary_path.read_text(encoding='utf-8'))['models']
}

rows = []
for label, no_key, gic_key in (
    ('Task4.2 L1500', 'L1500_no_gic', 'L1500_formal_gic'),
    ('Task4.2 L1000', 'L1000_no_gic', 'L1000_formal_gic'),
):
    no_value = float(task42[no_key]['parameters']['fnl_loc']['median'])
    gic_value = float(task42[gic_key]['parameters']['fnl_loc']['median'])
    sigma = float(task42[gic_key]['maximum_posterior_sample']['sigma_w2'])
    rows.append((label, no_value, gic_value, gic_value - no_value, sigma))

no_value = float(task43_models['no_gic']['fnl_loc']['q50'])
gic_value = float(task43_models['formal_gic']['fnl_loc']['q50'])
sigma = float(task43_models['formal_gic']['formal_gic']['sigma_w2_map'])
rows.append(('Task4.3 smin=50', no_value, gic_value, gic_value - no_value, sigma))

table = [
    '| case | no GIC fNL | 当前单项 GIC fNL | 中心位移 | sigma_W^2 |',
    '|---|---:|---:|---:|---:|',
]
for label, no_value, gic_value, shift, sigma in rows:
    table.append(f'| {label} | {no_value:.2f} | {gic_value:.2f} | {shift:+.2f} | {sigma:.3e} |')

display(Markdown('\n'.join(table)))

| case | no GIC fNL | 当前单项 GIC fNL | 中心位移 | sigma_W^2 |
|---|---:|---:|---:|---:|
| Task4.2 L1500 | 75.48 | 82.09 | +6.60 | 7.533e-05 |
| Task4.2 L1000 | 63.63 | 86.36 | +22.73 | 2.095e-04 |
| Task4.3 smin=50 | 3.56 | 7.42 | +3.86 | 2.227e-05 |

这些结果说明：单常数修正确实会移动 posterior，且体积越小通常越明显。它们仍然是有用的 baseline，但它们只检验了当前近似模型，而不是论文完整式 (2.20)。

还要注意，Task4.2 和 Task4.3 都用 ensemble-mean data vector 压低 realization noise，却用 single-realization covariance 报告单个 survey 的误差。因此非常小的 $\chi^2$ 或“结果与真值一致”不足以证明遗漏的 shape term 为零；当前 closure test 对小模型差异的鉴别力本来就有限。

## 7. Task4.2：缺什么，风险有多大？

Task4.2 是均匀 cubic subbox，random 也在每个 cube 内均匀生成。因此它主要是 global IC，没有 Task4.3 那种 data-derived radial selection。

对 Task4.2，当前缺失的核心是 $B(s)$ 的显式计算：

$$
\xi_{\rm exact}(s)=\xi(s)-2B(s)+C
\quad\text{vs}\quad
\xi_{\rm current}(s)=\xi(s)-C.
$$

如果 separation 远小于 window 的变化尺度，$B(s)\approx C$ 可能很好。但当前最大尺度为 350 Mpc/h：

$$
\frac{s_{\max}}{L_{1500}}\simeq0.23,
\qquad
\frac{s_{\max}}{L_{1000}}=0.35.
$$

特别是 L1000，不能只依据论文的 CMASS 示例就假定交叉项一定可忽略。local PNG 又增强了低-k 模式，而 GIC 正是对这些模式最敏感，所以应做 matched full-vs-single-term A/B。

## 8. Task4.3：还有一个更明确的 RIC 问题

Task4.3 的 random catalog 不是使用独立、预先给定的平滑 $n(z)$。它从每个 phase 的 halo redshift 中有放回重采样：

$$
\text{random radial policy}=\text{data redshift resample}.
$$

在 de Mattia & Ruhlmann-Kleider (2019) 的术语中，这就是 shuffled scheme。它不仅强制整个 survey 的平均密度为零，还会在每个径向层中压低由数据估计出来的 radial modes，因此引入 radial integral constraint：

$$
\xi_{\rm RIC}
=
\xi^{\rm c}
-IC^{\delta,{\rm rad}}
-IC^{{\rm rad},\delta}
+IC^{{\rm rad},{\rm rad}}.
$$

当前的一维 global $W_2(k)$ 没有 $\epsilon_{\rm rad}(r,x)$ 这个“是否处于同一 radial bin”的信息，所以不能表示完整 RIC。

In [3]:
# 代码大纲：
# 1. 读取 Task4.3 正式 manifest 的第一行，它代表 ph000 的实际数据生成口径。
# 2. 打印 geometry、redshift 范围与 random radial policy。
# 3. 用断言确认当前生产数据确实属于 data-redshift-resample，也就是论文中的 shuffled 类别。

manifest_path = ROOT / 'outputs/task43_outputs/manifests/task43_mmin1p4e13_x25.jsonl'
first_row = json.loads(manifest_path.read_text(encoding='utf-8').splitlines()[0])

print('phase                  =', first_row['phase'])
print('redshift range         =', (first_row['zmin'], first_row['zmax']))
print('random radial policy   =', first_row['random_radial_policy'])
print('random multiplier      =', first_row['random_multiplier'])

assert first_row['random_radial_policy'] == 'data_redshift_resample'
print('\n检查通过：Task4.3 测量不仅有 global IC，也具有 data-derived n(z) 引入的 RIC。')

phase                  = ph000
redshift range         = (0.6, 0.8)
random radial policy   = data_redshift_resample
random multiplier      = 25

检查通过：Task4.3 测量不仅有 global IC，也具有 data-derived n(z) 引入的 RIC。


Task4.3 的 radial shell 厚度约为 $1936-1535\simeq401$ Mpc/h，而拟合一直到 $s=350$ Mpc/h。也就是说，最大 separation 与径向 window 厚度相当，边界和 radial-mode subtraction 不应在没有 A/B 的情况下被视为纯常数。

此外，当前 full25 mean 使用 25 个 phase 各自 data-resampled 的 random，但正式 $W_2$ cache 来自 ph000 random。即使只研究 global $C$，也仍需检查“ph000 window”能否代表 25 个 realization-dependent radial windows。

## 9. 哪些问题是主要的，哪些是次要但仍需记录的？

### 当前最主要的两项

1. **完整 GIC cross terms 未显式计算**：当前用 $-C$ 近似 $-2B(s)+C$；
2. **Task4.3 RIC 未建模**：measurement 使用 data-derived random $n(z)$，theory 却只有 global scalar $C$。

### 做 science-ready 声明前还应检查

1. 论文指出 Poisson/contact shot noise 经过 IC window 后可以泄漏到非零 separation；固定 2PCF 的 $sn_0=0$ 并不自动证明这部分严格为零；
2. Landy–Szalay bin 的精确理论平均应使用 bin 内实际 RR weighting，而当前主线使用 volume-shell average；10 Mpc/h 窄 bin 下可能很小，但应量化；
3. Task4.3 是 real-space monopole，所以 RSD wide-angle 不是此处首要问题；到了 Task4.4 的 RSD multipoles，window multipole mixing 和 wide-angle 项会更重要。

## 10. 常见问题

### Q1：所以我们以前的结果错了吗？

不能直接说错。当前结果正确回答了“如果采用 parameter-dependent single-term GIC approximation，posterior 是什么”。问题是它还没有回答“采用论文完整三项 GIC，posterior 是否仍然相同”。

### Q2：既然 W2 来自真实 random，为什么还不完整？

因为 RR pair separation distribution 足以计算平均模式的方差 $C$，但 density–mean cross term $B(s)$ 还需要知道：当一对点被固定在 separation $s$ 时，其中一个点与整个 survey 其他位置如何相关。这是依赖 $s$ 的三点 window 信息，不包含在一条 $W_2(k)$ 曲线中。

### Q3：Landy–Szalay 不是已经除掉 survey geometry 了吗？

它用 RR 很好地消除了普通边界 pair loss，但平均密度仍然是从有限数据估计的，所以 integral constraint 不会因为除以 RR 自动消失。

### Q4：为什么论文说单项近似很好，我们还要检查？

论文证明它在特定 CMASS-like case 中数值上很好，而不是证明它对任意 geometry、任意尺度和 local PNG 都严格成立。我们的 L1000 cube、窄 radial lightcone 和低-k PNG sensitivity 都与那个示例不同。

## 11. 下一步怎样做才是严格的 matched A/B？

### Task4.2：先做最干净的 full-GIC 验证

1. 保持相同数据、covariance、FullDiscrete modes、PNG template、参数和 fit range；
2. 对 L1500 和 L1000 cube 计算 $B(s)$；
3. 比较 $\xi-C$ 与 $\xi-2B(s)+C$；
4. 报告 $\Delta\chi^2$、最大 residual/covariance sigma 和 $\Delta f_{\rm NL}$。

### Task4.3：必须把 GIC-only 与 RIC 分开

1. 用独立平滑或 uniform-comoving-volume $n(z)$ 的 random 重新测量一条 GIC-only branch；
2. 对现有 data-redshift-resample branch 构建论文的 radial kernels，计算 full RIC；
3. 检查每个 phase 的 radial window，不能只用 ph000 代表全部 phase 而不做验证；
4. 最后比较 no IC、single-term GIC、full GIC、full RIC 四种严格受控结果。

在这些 A/B 完成前，最稳妥的命名是 **parameter-dependent single-term GIC approximation**，而不是 complete/formal GIC。

## 12. 明天报告可以直接使用的英文表述

> **We model the leading global integral-constraint contribution using a parameter-dependent RR-window constant offset, $\xi_0^{\rm obs}(s)\simeq\xi_0(s)-\sigma_W^2(\theta)$. This corresponds to the conventional single-term approximation, rather than the complete density–IC cross-term formalism of de Mattia & Ruhlmann-Kleider (2019).**

如果要同时说明 Task4.3 的限制，可以再加一句：

> **Because the Task4.3 random redshifts are resampled from each data realization, a radial integral constraint is also present and remains to be modelled explicitly.**

## 13. 可追溯文件

- 论文：[de Mattia & Ruhlmann-Kleider (2019), arXiv:1904.08851](https://arxiv.org/abs/1904.08851)
- Task4.2 当前模型：codes/task4/task4p2_pk_rawbox_subbox/task4p2_fit_2pcf_shellavg_fixed_sn0.py
- Task4.3 当前模型：codes/task43/task43_fit_minimal_closure.py
- Task4.3 random 构造：codes/task43/task43_build_randoms.py
- Task4.3 实际 manifest：outputs/task43_outputs/manifests/task43_mmin1p4e13_x25.jsonl
- 项目已有文献笔记：observelearn/note/GIC_literature_review.md
- 当前任务书风险记录：agent/task.md 的 Task4.3 小节

---

### 最终记忆卡片

$$
\boxed{
\text{完整：}\;\xi-2B(s)+C
\qquad
\text{当前：}\;\xi-C
\qquad
\text{近似条件：}\;B(s)\simeq C
}
$$

Task4.3 还要额外记住：

$$
\boxed{\text{data-resampled }n(z)\;\Longrightarrow\;\text{RIC，不只是 GIC。}}
$$

## 14. 2026-07-12 更新：路线 A 的 radial single-term RIC 已实现

前面的第 11--12 节记录的是实现前的风险判断。本节记录现在真正采用并跑完的口径，优先级更高。

我们**没有**改测量、FKP、covariance、fit range 或 mother-box support，也**没有**声称实现论文完整三项 RIC。当前模型固定为

$$
\boxed{\xi_0^{\rm single,RIC}(s;\theta)=\xi_0^{\rm noIC}(s;\theta)-IC_0^{\rm rad,rad}(s;\theta)}.
$$

其中两个 density--RIC cross terms 仍然省略；radial normalization 已自动包含 global normalization，所以不再额外减一次 $\sigma_W^2$。$P(k)$ 侧同样只加入这个 radial response，不另加 standalone GIC：

$$
\mathbf P_{\rm obs}=\mathbf W_{\rm geom}\mathbf P_{\rm theory}-\mathbf W_{\rm RIC}\mathbf P_{\rm theory}.
$$

### 为什么最终没有继续用朴素 quadruplet MC？

最初的四点 MC 被极少数很近的 inner pair 主导，50k 到 200k 仍会出现重尾噪声。Task4.3 random 的生成式严格把 data-resampled radial redshift 与 uniform positive-octant direction 独立抽样，所以选择函数可分离：

$$W(r,\Omega)=W_{\rm rad}(r)W_{\rm octant}(\Omega).$$

因此同一个四点 radial-auto kernel 可无偏地压缩为

$$p(s,\Delta)=\sum_{a\le b}p_{ab}H_{ab}(s)H_{ab}(\Delta),$$

这里 $a,b$ 是 radial bins，$H_{ab}$ 由一次 $2^{22}$ Sobol 正八分体角向积分得到；outer marginal 再严格匹配 pycorr weighted RR。这个 factorization 去掉了重尾 MC 噪声，但没有改变 single-term RIC 的物理定义。2PCF 与 $P(k)$ 都从这一个 cache 投影。

In [4]:
# 代码大纲：
# 1. 读取路线 A 的权威总审计，而不是手工复制运行日志。
# 2. 显示全部数值 gate、2PCF/P(k) 主 posterior 与 correction/误差大小。
# 3. 用断言保证 notebook 展示的是已经通过完整审计的最终产物。

ric_audit_path = ROOT / 'outputs/task43_outputs/ric_singleterm/audits/task43_ric_singleterm_audit.json'
ric = json.loads(ric_audit_path.read_text(encoding='utf-8'))

gate_lines = ['| gate | pass |', '|---|---:|']
for name, passed in ric['gates'].items():
    gate_lines.append(f'| {name} | {passed} |')
display(Markdown('### Numerical gates\n\n' + '\n'.join(gate_lines)))

posterior_lines = [r'| probe/model | $f_{\rm NL}$ (16/50/84%) |', '|---|---:|']
for model, row in ric['main_posteriors']['2pcf'].items():
    posterior_lines.append(f"| 2PCF {model} | {row['fnl_q50']:.2f} [{row['fnl_q16']:.2f}, {row['fnl_q84']:.2f}] |")
for model, row in ric['main_posteriors']['pk'].items():
    posterior_lines.append(f"| P(k) {model} | {row['fnl_q50']:.2f} [{row['fnl_q16']:.2f}, {row['fnl_q84']:.2f}] |")
display(Markdown('### Fixed-data posterior comparison\n\n' + '\n'.join(posterior_lines)))

display(Markdown(
    f"Baseline correction size: **{ric['baseline_correction']['xi_max_abs_over_sigma']:.3f} sigma** "
    f"per 2PCF bin and **{ric['baseline_correction']['pk_max_abs_over_sigma']:.3f} sigma** "
    "per P(k) bin."
))
assert ric['gates']['all_required_numerical_gates'] is True


### Numerical gates

| gate | pass |
|---|---:|
| all_required_numerical_gates | True |
| bin_width_model_pk | True |
| bin_width_model_xi | True |
| global_limit | True |
| hankel_consistency | True |
| main_chains | True |
| parameter_dependence | True |
| phase_dependence | True |
| posterior_convergence_all | True |
| random_50k_to_200k_model_pk | True |
| random_50k_to_200k_model_xi | True |
| ric_induced_covariance_negligible | True |

### Fixed-data posterior comparison

| probe/model | $f_{\rm NL}$ (16/50/84%) |
|---|---:|
| 2PCF formal_gic | 7.15 [-21.28, 33.84] |
| 2PCF no_gic | 3.87 [-23.63, 29.62] |
| 2PCF radial_singleterm | 10.18 [-17.95, 37.35] |
| P(k) geometry_only | -3.86 [-35.33, 25.66] |
| P(k) radial_singleterm | -0.54 [-32.16, 28.92] |

Baseline correction size: **0.072 sigma** per 2PCF bin and **0.109 sigma** per P(k) bin.

## 15. 怎样解释最终结果？

1. radial correction 确实不是常数：在 fiducial $f_{\rm NL}=0,b_1=2.5$ 下，2PCF correction 从约 $1.20\times10^{-4}$ 随 $s$ 下降到 $2.01\times10^{-5}$。
2. 但它相对当前单份 lightcone 误差较小：逐 bin 最大约 $0.072\sigma$（2PCF）和 $0.109\sigma$（$P(k)$）。
3. 2PCF 中 no IC、旧 global single-term、新 radial single-term 分别给出 $f_{\rm NL}=3.87,7.15,10.18$ 的中位数；$P(k)$ geometry-only 与 radial single-term 分别为 $-3.86,-0.54$。所有结果仍与真值 0 相容，2PCF--$P(k)$ 也相容。
4. $\Delta\chi=1/2/4$、50k 到 200k、代表 phase、global limit、Hankel/direct、$f_{\rm NL}=0,\pm100$ 与 free-$sn_0$ gate 全部通过。
5. phase-dependent RIC correction 引入的额外 variance 远小于现有 covariance，因此本轮继续固定 covariance；这并不消除原 analytic covariance 自身已有的 diagnostic caveat。

最重要的表述边界仍然是：**这是经过收敛验证的 radial single-term approximation，不是 de Mattia & Ruhlmann-Kleider (2019) 的完整三项 RIC。**

## 16. 路线 A 的最终可追溯入口

- factorized kernel：`codes/task43/task43_build_ric_factorized_kernel.py`
- 共用 operator：`codes/task43/task43_make_ric_factorized_operator.py`
- 2PCF fit：`codes/task43/task43_fit_minimal_closure.py` 的 `radial_singleterm` branch
- $P(k)$ fit：`codes/task43/task43_fit_pk_lightcone.py --ric-operator ...`
- 权威总审计：`outputs/task43_outputs/ric_singleterm/audits/task43_ric_singleterm_audit.json`
- PDF 图：`plots/task43/ric_singleterm/`

报告中可使用：

> **We include a binned radial integral-constraint response using the conventional single-term approximation, $\xi_0^{\rm obs}\simeq\xi_0-IC_0^{\rm rad,rad}$, while keeping the measurements and covariances fixed. The density--RIC cross terms of the complete de Mattia & Ruhlmann-Kleider (2019) formalism are not included.**